src/gsy_e/models/area/scm_manager.py

```diff
@@ -0,0 +1,249 @@
+import logging
+from dataclasses import dataclass
+from typing import Dict
+from uuid import uuid4
+
+from gsy_framework.data_classes import Trade
+from pendulum import DateTime
+
+from gsy_e.constants import DEFAULT_GRID_SELLER_STRING, DEFAULT_SCM_SELLER_STRING
+
+
+@dataclass
```

Comment:
Can you please explain what the benefit is here for declaring a dataclass? If I look at the logic it does a lot of things on top of only holding data.

Reply:
Admittedly there is not much of a benefit here. I was originally planning to have this as a container of data, but shortly after it was evident that these data would benefit from methods that would contain logic too. Of course though, it does not hurt for this to be a dataclass, but if you would prefer it to be a normal class ping me here and I will adapt it, thanks!

Reply:
OK, thanks! Understood. Leave it as is then.